### ISSUE TITLE
Determine the path of the currently viewed file

### ISSUE THREAD

**Issue by nkolban** [CoatiSoftware/SourcetrailBugTracker#556]
With reference to the following screen shot:

![image](https://user-images.githubusercontent.com/3421732/36936978-0204aa1a-1ed2-11e8-942f-53a21546d759.png)

Notice that I have searched for a function and found it.  I also see the name of the source file.  In this case it is called `main.cpp`.  However, in my project I have many folders and each folder contains a `main.cpp`.  My desire is to *glance* and see *which* of the `main.cpp` files I am looking at.  I see that I can hover over the file name in the header and am rewarded with a full path ... however I'm hoping that we can have a mechanism (perhaps a user preference) to show the path in the title bar as well?

**Comment by egraether** [comment]
We had a quick discussion. The best solution is probably to check whether the same filename exists multiple times in the project and then show it with the parent directory that is different e.g.:
```
/dev/project/src/server/main.cpp  ->  server/main.cpp
/dev/project/src/client/main.cpp  ->  client/main.cpp
```

**Comment by nkolban** [comment]
Thanks for the super fast response.  My vote would be to keep it super simple and give the user (me) the option to show the whole path to the source file rather than just the single parent directory.  That may still not be unique if I have multiple source files with the same name down deeper paths from the root of my source.  Again, I'm just a single data point ... this may be a good candidate for a user poll and see what others may also think / desire.

**Comment by egraether** [comment]
We are hesitant towards adding too many preferences. And this use-case should work without extra configuration by the user.

Generally showing full paths is not a good idea. They can be extraordinary long, so we would need a character limit (based on available space) and just show the end of the path, but who knows wether that includes the part you are looking for.

So best is to show what is different, which could look this way for deeper paths:

```
/dev/project/server/src/main.cpp  ->  server/.../main.cpp
/dev/project/client/src/main.cpp  ->  client/.../main.cpp
```
or this
```
/dev/project/server/src/main.cpp  ->  server - main.cpp
/dev/project/client/src/main.cpp  ->  client - main.cpp
```

**Comment by TomSmartBishop** [comment]
I agree that it would be beneficial to display the directory structure (relative to the project roo). Especially in a large code base (usually you try to understand these and not a pet project) you might have no idea where the file is located even if the filename itself is unique

**Comment by egraether** [comment]
+1 vie mail

**Comment by egraether** [comment]
Implemented with 2019.3.13

Added option to Preferences to show directory of code file in code file title relative to project file.
e.g. `src/app - main.cpp`.

If path is too long it gets elided at the front e.g. `...ery_long_dir_name/src/app - main.cpp`
